tests/system/test_multigroups/test_multigroups.py

```diff
@@ -0,0 +1,259 @@
+    expected_mtime_changed = [os.path.join(group_path, filename), os.path.join(group_path, 'merged.mg')]
+    expected_mtime_changed.extend([os.path.join(mg_folder_path, calculate_mg_name(groups)) for _, groups
+                                   in agent_groups.items() if target_group in groups])
+
+    sleep(time_to_sync)
+    new_host_files = get_mtime(folders_to_check)
+
+    for host, files in new_host_files.items():
+        for file, mtime in files.items():
+            for new_mtime_file in expected_mtime_changed:
+                if new_mtime_file in file:
+                    try:
+                        assert mtime != host_files[host][file], f"This file should have changed in {host}: {file}"
+                    except KeyError:
+                        assert os.path.basename(file) == filename
+                        assert group_content == host_manager.run_command(host, f"cat {file}"), \
+                            f"The new file content is not the expected in {file}"
+                    break
+            else:
+                assert mtime == host_files[host][file], f"This file changed its modification time in {host}: {file}"
+
+
+def test_multigroups_deleted(clean_files, agent_healthcheck, create_multigroups):
+    """Check that multigroups are removed when expected.
+
+    Unassign an agent from their groups or delete the groups. Check that the associated
+    multigroup disappears in both cases.
+    """
+    for agent_name, groups in agent_groups.items():
+        # Check that multigroups exists for each agent.
+        mg_path = os.path.join(mg_folder_path, calculate_mg_name(groups))
+        assert '' != host_manager.run_shell(test_hosts[0], f"ls {mg_path}"), f"{mg_path} should exist, but it does not."
+        agent_id = get_agent_id(token=token, agent_name=agent_name)
+
+        for group in groups:
+            if group != 'default':
+                if agent_name == 'wazuh-agent1':
+                    # Unassign agent.
+                    response = host_manager.make_api_call(host=test_hosts[0], token=token, method='DELETE',
+                                                          endpoint=f"/agents/{agent_id}/group/{group}")
+                    assert response['status'] == 200, f"Failed when unassigning {agent_name} agent from " \
+                                                      f"group {group}: {response}"
+                else:
+                    # Delete group.
+                    response = host_manager.make_api_call(host=test_hosts[0], token=token, method='DELETE',
+                                                          endpoint=f"/groups?groups_list={group}")
+                    assert response['status'] == 200, f"Failed to delete {group} group: {response}"
+
+        # Check that multigroups no longer exists for each agent.
+        sleep(time_to_update)
+        assert '' == host_manager.run_shell(test_hosts[0], f"ls {mg_path}"), f"{mg_path} should not exist, but it does."
```

Comment:
We should check that the folder doesn't exist as in the requested change in `create_multigroups`